sycl/doc/extensions/proposed/sycl_ext_codeplay_cuda_cluster_group.asciidoc

```diff
@@ -0,0 +1,413 @@
+    }
+----
+
+
+[%header,cols="5,5"]
+|===
+|Method
+|Description
+
+|`linear_id get_group_linear_id() const`
+|Returns the linearized id of the calling work-group within the cluster
+
+|`linear_id get_local_linear_id() const`
+|Returns the linearized index of the calling work-item within the cluster
+
+|`range_type get_group_range() const`
+|Returns the number of work-groups in each dimension within the cluster
+
+|`id_type get_group_id() const`
+|Returns the id of the calling work-group along each dimension within the cluster
+
+|`id_type get_local_id() const`;
+|Returns the id of calling work-item along each dimension within the cluster
+
+|`range_type get_local_range() const`;
+|Returns the number of work-items along each dimension within the cluster.
+
+|`linear_id_type get_local_linear_range() const`;
+|Returns a linearized version of the `range_type` returned by  `get_local_range`
+
+|`linear_id_type get_group_linear_range() const`;
+|Returns a linearized version of the `range_type` returned by  `get_group_range`
+
+|`bool leader() const`;
+|Returns true for exactly one work-item in the cluster, if the calling work-item
+is the leader of the cluster group. The leader is guaranteed to be the work-item
+for which `get_local_linear_id` return 0.
+|===
+
+
+== Accessing another work-group's local memory
+
+Work-groups within the cluster have the ability to access another work-group's 
+local memory. Typically addresses which reside in the local memory of a 
+work-group can only be accessed by the work-items of that work-group. 
+Therefore, to access another work-group's local memory, the address needs to be
+mapped such that the address in another work-group is accessible within the 
+calling work-item. Further, to access another work-group's local memory,
+all the work-groups within the cluster must exist and the work-groups should
+not cease to exist before all the memory operations are completed. This can be 
+ensured by synchronizing all the work-items within the cluster before and after
+the local memory operations, using `group_barrier`.
+
+A member function of the `cluster_group` class; 
+`map_cluster_local_pointer` will perform the mapping and return a pointer 
+which can then be dereferenced by the calling work-item. 
+
+
+[%header,cols="10,5"]
+|===
+|Method
+|Description
+
+|T*  map_cluster_local_pointer(T* addr, size_t group_id)
+|Accepts the equivalent address to the memory location relative to the calling 
+work-item which is to be mapped from the local memory of the work-group, as 
+specified by the group_id, denoting the linear group_id within the cluster
+|===
+
+Conversely, `get_linear_cluster_group_id_for_local_pointer` will return the 
+linearized id of the work-group a mapped local memory address belongs to.
+
+[%header,cols="10,5"]
+|===
+|Method
+|Description
+
+|size_t  get_linear_cluster_group_id_for_local_pointer(T* addr)
+|Accepts a mapped pointer to the local memory of another work-group and returns
```

Comment:
We should specify what is returned in the case that the pointer has not been mapped, I assume since this pointer would be a regular local memory pointer, the work-group id would be that of the current work-item.

Reply:
Good point, 
I will change the wording to it accepting a valid shared memory address,
 and also add that any other address other than shared memory leads to UB

Reply:
Done in [824af1e](https://github.com/intel/llvm/pull/13594/commits/824af1e30203f42612a8c13bbe4740c91aeb1a2f)